src/Features/LanguageServer/Microsoft.CommonLanguageServerProtocol.Framework/Microsoft.CommonLanguageServerProtocol.Framework.Shared.projitems

```diff
@@ -0,0 +1,45 @@
+<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <PropertyGroup>
+    <MSBuildAllProjects Condition="'$(MSBuildVersion)' == '' Or '$(MSBuildVersion)' &lt; '16.0'">$(MSBuildAllProjects);$(MSBuildThisFileFullPath)</MSBuildAllProjects>
+    <HasSharedItems>true</HasSharedItems>
+    <SharedGUID>64eaded3-4b5d-4431-bbe5-a4aba1c38c00</SharedGUID>
+  </PropertyGroup>
+  <PropertyGroup Label="Configuration">
+    <Import_RootNamespace>Microsoft.CommonLanguageServerProtocol.Framework.Shared</Import_RootNamespace>
+  </PropertyGroup>
+  <ItemGroup>
+    <Compile Include="$(MSBuildThisFileDirectory)AbstractHandlerProvider.cs" />
+    <Compile Include="$(MSBuildThisFileDirectory)AbstractLanguageServer.cs" />
+    <Compile Include="$(MSBuildThisFileDirectory)AbstractLspLogger.cs" />
+    <Compile Include="$(MSBuildThisFileDirectory)AbstractRequestContextFactory.cs" />
+    <Compile Include="$(MSBuildThisFileDirectory)AbstractRequestScope.cs" />
+    <Compile Include="$(MSBuildThisFileDirectory)AbstractTelemetryService.cs" />
+    <Compile Include="$(MSBuildThisFileDirectory)HandlerProvider.cs" />
+    <Compile Include="$(MSBuildThisFileDirectory)Handlers\InitializedHandler.cs" />
+    <Compile Include="$(MSBuildThisFileDirectory)Handlers\InitializeHandler.cs" />
+    <Compile Include="$(MSBuildThisFileDirectory)IHandlerProvider.cs" />
+    <Compile Include="$(MSBuildThisFileDirectory)IInitializeManager.cs" />
+    <Compile Include="$(MSBuildThisFileDirectory)ILifeCycleManager.cs" />
+    <Compile Include="$(MSBuildThisFileDirectory)ILspLogger.cs" />
+    <Compile Include="$(MSBuildThisFileDirectory)ILspServices.cs" />
+    <Compile Include="$(MSBuildThisFileDirectory)IMethodHandler.cs" />
+    <Compile Include="$(MSBuildThisFileDirectory)INotificationHandler.cs" />
+    <Compile Include="$(MSBuildThisFileDirectory)IQueueItem.cs" />
+    <Compile Include="$(MSBuildThisFileDirectory)IRequestContextFactory.cs" />
+    <Compile Include="$(MSBuildThisFileDirectory)IRequestExecutionQueue.cs" />
+    <Compile Include="$(MSBuildThisFileDirectory)IRequestHandler.cs" />
+    <Compile Include="$(MSBuildThisFileDirectory)ITextDocumentIdentifierHandler.cs" />
+    <Compile Include="$(MSBuildThisFileDirectory)LanguageServerConstants.cs" />
+    <Compile Include="$(MSBuildThisFileDirectory)LanguageServerEndpointAttribute.cs" />
+    <Compile Include="$(MSBuildThisFileDirectory)QueueItem.cs" />
+    <Compile Include="$(MSBuildThisFileDirectory)RequestExecutionQueue.cs" />
+    <Compile Include="$(MSBuildThisFileDirectory)RequestHandlerMetadata.cs" />
+    <Compile Include="$(MSBuildThisFileDirectory)RequestShutdownEventArgs.cs" />
+    <Compile Include="$(MSBuildThisFileDirectory)WrappedHandlerProvider.cs" />
+    <Compile Include="..\..\..\Compilers\Core\Portable\InternalUtilities\IsExternalInit.cs" Link="Utilities\IsExternalInit.cs" />
+  </ItemGroup>
+  <ItemGroup>
+    <Content Include="$(MSBuildThisFileDirectory)BannedSymbols.txt" />
```

Comment:
Is this file needed? It's contents are very Roslyn specific, and I would imagine the consuming projects would be the place to put it. Not that I expect it will cause any harm either 🤷‍♂️

Reply:
Seems a bit weird to me too

Reply:
yeah, I'll remove this.  Shouldn't be necessary here.